src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java

```diff
@@ -197,7 +197,6 @@ public int[] getRequiredTokens() {
 
     @Override
     public void beginTree(DetailAST rootAST) {
-        pkgName = null;
```

Comment:
As all reset before each file. We need test with the two files execution in one test.
First file with package, second is without package.
Second file should try to trigger violation related to same package name.

Reply:
It is hard to find this in real life, because usually all have packages.
But our contract that all files randomly processed, without any context should affect any other file.

Reply:
https://github.com/checkstyle/checkstyle/pull/13284#discussion_r1243120846 I have tried this. but I will give time once again to this

Reply:
@romani  Done. 
Thanks :)